### PR TITLE
feat: add edit charge component.

### PR DESCRIPTION
**Pull request by BLasan**
Description
Created Edit-Charge component for editing the charges selected

Related issues and discussion
#586

Screenshots, if any
Screenshot from 2020-02-23 18-15-06

Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

[] If you have multiple commits please combine them into one commit by squashing them.

[] Read and understood the contribution guidelines at web-app/.github/CONTRIBUTING.md.

### CODE REVIEW

File: src/app/products/charges/view-charge/view-charge.component.ts [hunk 1]
```diff
@@ -34,7 +34,9 @@ export class ViewChargeComponent implements OnInit {
               private router: Router,
               private dialog: MatDialog) {
     this.route.data.subscribe((data: { charge: any }) => {
+      console.log(data);
```
Comment (spider0061) [inline src/app/products/charges/view-charge/view-charge.component.ts]: remove this.

File: src/environments/environment.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export const environment = {
   production: false,
   version: env.npm_package_version + '-dev',
   fineractPlatformTenantId: 'default',  // For connecting to server running elsewhere update the tenant identifier
-  baseApiUrl: 'https://demo.openmf.org',  // For connecting to server running elsewhere update the base API URL
+  baseApiUrl: 'https://demo.mifos.io/',  // For connecting to server running elsewhere update the base API URL
```
Comment (spider0061) [inline src/environments/environment.ts]: do not include environment.ts in your PR.

File: package.json [hunk 3]
```diff
@@ -21,8 +21,7 @@
     "test:ci": "npm run env -s && npm run lint -s && ng test --code-coverage --watch=false",
     "e2e": "npm run env -s && ng e2e",
     "translations:extract": "ngx-translate-extract --input ./src --output ./src/translations/template.json --format=json --clean -sort --marker extract",
-    "docs": "hads ./docs -o",
-    "env": "ngx-scripts env npm_package_version"
+    "docs": "hads ./docs -o"
```
Comment (spider0061) [inline package.json]: Do not include package.json and package-lock.json in your PR.

File: src/app/products/charges/view-charge/view-charge.component.ts [hunk 4]
```diff
@@ -34,7 +34,9 @@ export class ViewChargeComponent implements OnInit {
...
       this.chargeData = data.charge;
+      console.log(this.chargeData);
```
Comment (spider0061) [inline src/app/products/charges/view-charge/view-charge.component.ts]: remove this.

File: src/app/products/charges/edit-charge/edit-charge.component.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  ngOnInit() {
+    this.createUserForm();
+  }
+
+  /**
+   * Creates the user form.
+   */
+  createUserForm() {
```
Comment (spider0061) [inline src/app/products/charges/edit-charge/edit-charge.component.ts]: the function name should be editChargeForm()

File: src/app/products/charges/edit-charge/edit-charge.component.html [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+          </div>
+        </div>
+      </form>
+    </mat-card-content>
+    <mat-card-actions fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center" fxLayoutGap="5px">
+      <button type="button" mat-raised-button [routerLink]="['../']">Cancel</button>
+      <button mat-raised-button color="primary" [disabled]="!form.valid" (click)="submit()">Submit</button>
+    </mat-card-actions>
+  </mat-card>
+</div>
```
Comment (spider0061) [inline src/app/products/charges/edit-charge/edit-charge.component.html]: add an empty line here.

File: src/app/products/charges/edit-charge/edit-charge.component.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      this.selectedTime = this.chargeData.chargeTimeType.id;
+    });
+  }
+
+  ngOnInit() {
+    this.createUserForm();
+  }
+
+  /**
+   * Creates the user form.
```
Comment (spider0061) [inline src/app/products/charges/edit-charge/edit-charge.component.ts]: the comment should be creates the edit charge form.

File: src/app/products/products.service.ts [hunk 8]
```diff
@@ -112,6 +112,24 @@ export class ProductsService {
     return this.http.get(`/charges/${chargeId}`);
   }
 
+  getSelectedCharge(selectedCharge: string, template: boolean = false): Observable<any> {
```
Comment (spider0061) [inline src/app/products/products.service.ts]: What is the need to define a new function? You can get the charge data through getCharge() function too.

File: src/app/products/charges/edit-charge/edit-charge.component.html (hunk 6 above)
Comment (BLasan) [reply src/app/products/charges/edit-charge/edit-charge.component.html]: In which line should I add ?

Comment (spider0061) [reply src/app/products/charges/edit-charge/edit-charge.component.html]: at the end .

**Comment by abhi40308** [comment]
Hi, try to update the same pr only instead of closing and sending a new one.

**Pull request closed without merging**